lighthouse-core/scripts/i18n/collect-strings.js

```diff
@@ -30,18 +28,337 @@ const ignoredPathComponents = [
   '-renderer.js',
 ];
 
+
 // @ts-ignore - @types/esprima lacks all of these
-function computeDescription(ast, property, startRange) {
+function computeDescription(ast, property, value, startRange) {
   const endRange = property.range[0];
+
   for (const comment of ast.comments || []) {
     if (comment.range[0] < startRange) continue;
     if (comment.range[0] > endRange) continue;
-    return comment.value.replace('*', '').trim();
+    if (comment.value.includes('@')) {
+      // This is a complex description with description and examples.
+      let description = '';
+      /** @type {Record<string, string>} */
+      const examples = {};
+
+      const r = /@(\w+) ({.+})?(.*)(\n|$)/g;
+      let matches;
+      while ((matches = r.exec(comment.value)) !== null) {
+        const tagName = matches[1];
+        const example = matches[2];
+        const messageOrPlaceholder = matches[3].trim();
+
+        if (tagName === 'description') {
+          description = messageOrPlaceholder;
+        } else if (tagName === 'example') {
+          examples[messageOrPlaceholder] = example.substring(1, example.length - 1);
+        }
+      }
+
+      // Make sure description is not empty
+      if (description.length === 0) throw Error(`Empty @description for message "${value}"`);
+      return {description, examples};
+    }
+
+    const description = comment.value.replace('*', '').trim();
+
+    // Make sure description is not empty
+    if (description.length === 0) throw Error(`Empty description for message "${value}"`);
+
+    // The entire comment is the description, so return everything.
+    return {description};
+  }
+  throw Error(`No Description for message "${value}"`);
+}
+
+/**
+ * Take a series of LHL format ICU messages and converts them
+ * to CTC format by replacing {ICU} and `markdown` with
+ * $placeholders$. Functional opposite of `bakePlaceholders`. This is commonly
+ * called as one of the first steps in translation, via collect-strings.js.
+ *
+ * Converts this:
+ * messages: {
+ *  "lighthouse-core/audits/seo/canonical.js | explanationDifferentDomain" {
+ *    "message": "Points to a different domain ({url})",
+ *    },
+ *  },
+ * }
+ *
+ * Into this:
+ * messages: {
+ *  "lighthouse-core/audits/seo/canonical.js | explanationDifferentDomain" {
+ *    "message": "Points to a different domain ($ICU_0$)",
+ *    "placeholders": {
+ *      "ICU_0": {
+ *        "content": "{url}",
+ *        "example": "https://example.com/"
+ *      },
+ *    },
+ *  },
+ * }
+ *
+ * Throws if the message violates some basic sanity checking.
+ *
+ * @param {string} message
+ * @param {Record<string, string>} examples
+ * @return {ICUMessageDefn}
+ */
+function convertMessageToPlaceholders(message, examples = {}) {
```

Comment:
`convertMessageToCtc`?

I understand if you're attached to the name by now, though :)